accounts/abi/event_test.go

```diff
@@ -54,3 +58,61 @@ func TestEventId(t *testing.T) {
 		}
 	}
 }
+
+type testResult struct {
+	Value1 *big.Int
+	Value2 *big.Int
+}
+type testCase struct {
+	definition string
+	want       testResult
+}
+
+func (tc testCase) encoded() []byte {
+	var b bytes.Buffer
+	if tc.want.Value1 != nil {
+		b.Write(math.PaddedBigBytes(math.U256(tc.want.Value1), 32))
+	}
+	if tc.want.Value2 != nil {
+		b.Write(math.PaddedBigBytes(math.U256(tc.want.Value2), 32))
+	}
+	return b.Bytes()
+}
+
+func TestEventUnpack(t *testing.T) {
+
+	table := []testCase{
+		{
+			definition: `[{"anonymous":false,"inputs":[{"indexed":true,"name":"value1","type":"uint256"},{"indexed":false,"name":"value2","type":"uint256"}],"name":"transfer","type":"event"}]`,
+			want:       testResult{Value2: big.NewInt(10)},
+		},
+		{
+			definition: `[{"anonymous":false,"inputs":[{"indexed":false,"name":"value1","type":"uint256"},{"indexed":false,"name":"value2","type":"uint256"}],"name":"transfer","type":"event"}]`,
+			want:       testResult{Value1: big.NewInt(100), Value2: big.NewInt(1)},
+		},
+		{
+			definition: `[{"anonymous":false,"inputs":[{"indexed":false,"name":"value1","type":"uint256"},{"indexed":true,"name":"value2","type":"uint256"}],"name":"transfer","type":"event"}]`,
+			want:       testResult{Value1: big.NewInt(100)},
+		},
```

Comment:
A test that should be added for the sake of sanity would be in the case of a indexed static array type. Say for example I'm passing in an indexed uint[2], and a non indexed string, this should take into account the length of the static array and ignore it...perhaps this should be handled by another variable in the events.go file above to best handle this.